src/elements/date-picker/DatePicker.js

```diff
@@ -131,22 +142,71 @@ export class DatePicker {
 
     select(event, day, fireEvents) {
         swallowEvent(event);
-
-        this.selected = day.date;
-        this.updateHeading();
-
-        if (fireEvents) {
+        if (this.range) {
+            if (!this.selected && event) {
+                this.selected = day.date;
+            } else if (this.selected && this.selected2) {
+                this.selected = day.date;
```

Comment:
`selected` and `selected2` is there start/end? Can they be a bit more descriptive?